R/statistic-covariance.R

```diff
@@ -120,25 +120,30 @@ dpCovariance <- setRefClass(
 
 dpCovariance$methods(
   initialize = function(mechanism, varType, n, epsilon, columns, rng=NULL, imputeRng=NULL, 
-                        intercept=FALSE, formula=NULL, delta=1e-5) {
+                        intercept=FALSE, formula=NULL) {
+    # NOTE THIS IS CURRENTLY NOTE SECURE DUE TO SENS AND EPSILON NOT DISTRIBUTED PROPERLY
     .self$name <- 'Differentially private covariance matrix'
-    .self$mechanism <- mechanism
-    .self$varType <- varType
-    .self$n <- checkNValidity(n)
-    .self$epsilon <- epsilon
-    .self$delta <- delta
-    .self$rng <- rng
+    .self$mechanism <- checkMechanism(mechanism, "mechanismLaplace")
+    .self$varType <- checkVariableType(varType, c('integer', 'double', 'numeric', 'logical'))  #NEED TO CHANGE TO ALLOW MULTIPLE VARTYPES  
+    .self$n <- checkN(n)  #NEED TO ADD THING SAYING HOW THIS IS N PER COLUMN
+    .self$epsilon <- checkEpsilon(epsilon) #NEED TO CHANGE
```

Comment:
change the function or change the use of the function in the statistic?

Reply:
kinda both, you have to split up global epsilon in covariance statistic. It's not relevant for this PR, relevant to eventual PR on covariance.